util/hint/hint_processor.go

```diff
@@ -435,6 +440,26 @@ func (p *BlockHintProcessor) getBlockOffset(blockName model.CIStr) int {
 	return -1
 }
 
+// ParseOffsetFromBlockName parse block offset from blockName
+func (p *BlockHintProcessor) ParseOffsetFromBlockName(blockName model.CIStr, curLevel int) int {
```

Comment:
What is the difference between this function and `getBlockOffset`?

Reply:
ParseOffsetFromBlockName is just for parse blockName,but the getBlockOffset is always process with traversing on QueryBlock,there some constraints for block offset that as output.